js/views/modals/EditListing.js

```diff
@@ -0,0 +1,529 @@
+      'change #editListingSlug': 'onChangeSlug',
+      'change .js-price': 'onChangePrice',
+      'change #inputPhotoUpload': 'onChangePhotoUploadInput',
+      'click .js-addPhoto': 'onClickAddPhoto',
+      'click .js-removeImage': 'onClickRemoveImage',
+      'click .js-cancelPhotoUploads': 'onClickCancelPhotoUploads',
+      ...super.events(),
+    };
+  }
+
+  get MAX_PHOTOS() {
+    return this.model.get('listing').get('item').maxImages;
+  }
+
+  onAddImage(image) {
+    const imageHtml = this.uploadPhotoT({
+      closeIconClass: 'js-removeImage',
+      ...image.toJSON(),
+    });
+
+    this.$photoUploadItems.append(imageHtml);
+  }
+
+  onRemoveImage(image, images, options) {
+    // 1 is added to the index to account for the .addElement
+    this.$photoUploadItems.find('li')
+      .eq(options.index + 1)
+      .remove();
+  }
+
+  onClickRemoveImage(e) {
+    // since the first li is the .addElement, we need to subtract 1 from the index
+    const removeIndex = $(e.target).parents('li').index() - 1;
+
+    this.images.remove(this.images.at(removeIndex));
+  }
+
+  onClickCancelPhotoUploads() {
+    this.inProgressPhotoUploads.forEach(photoUpload => photoUpload.abort());
+  }
+
+  onChangePrice(e) {
+    let updatedVal = $(e.target).val().trim();
+    const valAsNumber = Number(updatedVal);
+
+    if (!isNaN(valAsNumber)) {
+      const decimalPlaces = this.$currencySelect.val() === 'BTC' ? 8 : 2;
+      updatedVal = valAsNumber.toFixed(decimalPlaces);
+    }
+
+    $(e.target).val(updatedVal);
+  }
+
+  onChangeSlug(e) {
+    const val = $(e.target).val();
+
+    // we'll make the slug all lowercase,
+    // replace spaces with dashes and remove
+    // url unfreindly chars.
+    // todo: this could be made into a slugify utility
+    $(e.target).val(
+      val.toLowerCase()
+        .replace(/\s/g, '-')
+        .replace(/[^a-zA-Z0-9-]/g, '')
+        // replace consecutive dashes with one
+        .replace(/-{2,}/g, '-')
+    );
+  }
+
+  onChangeListingType(e) {
+    if (e.target.value !== 'PHYSICAL_GOOD') {
+      this.$conditionWrap.addClass('disabled');
+    } else {
+      this.$conditionWrap.removeClass('disabled');
+    }
+  }
+
+  onChangePhotoUploadInput() {
+    let photoFiles = Array.prototype.slice.call(this.$inputPhotoUpload[0].files, 0);
+
+    // prune out any non-image files
+    photoFiles = photoFiles.filter(file => file.type.startsWith('image'));
+
+    this.$inputPhotoUpload.val('');
+
+    const currPhotoLength = this.innerListing.get('item')
+      .get('images')
+      .length;
+
+    if (currPhotoLength + photoFiles.length > this.MAX_PHOTOS) {
+      photoFiles = photoFiles.slice(0, this.MAX_PHOTOS - currPhotoLength);
+
+      new SimpleMessage({
+        title: app.polyglot.t('editListing.errors.tooManyPhotosTitle'),
+        message: app.polyglot.t('editListing.errors.tooManyPhotosBody'),
+      })
+      .render()
+      .open();
+    }
+
+    if (!photoFiles.length) return;
```

Comment:
but if they exceed the max length, I strip out any about the max, which may be all of them. That's why I need the check after the potential stripping.

Eventually, I could update the UI, so the Add button is disabled if they are at the max. But, if they are at 9 and click add, they could still select more than 1, in which case I'll need to strip out all but 1.